This patch includes support for OVS port names > 16 on Solaris.

This patch has not been proposed upstream because we are not yet
proposing Solaris specific requirements upstream.

diff --git a/lib/netdev-vport.h b/lib/netdev-vport.h
index 048aa6ebf..f3553a84d 100644
--- a/lib/netdev-vport.h
+++ b/lib/netdev-vport.h
@@ -43,10 +43,12 @@ void netdev_vport_inc_tx(const struct netdev *,
 bool netdev_vport_is_vport_class(const struct netdev_class *);
 const char *netdev_vport_class_get_dpif_port(const struct netdev_class *);
 
-#ifndef _WIN32
-enum { NETDEV_VPORT_NAME_BUFSIZE = 16 };
-#else
+#ifdef __sun
+enum { NETDEV_VPORT_NAME_BUFSIZE = 96 };
+#elif _WIN32
 enum { NETDEV_VPORT_NAME_BUFSIZE = 256 };
+#else
+enum { NETDEV_VPORT_NAME_BUFSIZE = 16 };
 #endif
 const char *netdev_vport_get_dpif_port(const struct netdev *,
                                        char namebuf[], size_t bufsize)
diff --git a/lib/tnl-neigh-cache.c b/lib/tnl-neigh-cache.c
index 499efffbf..73418c7df 100644
--- a/lib/tnl-neigh-cache.c
+++ b/lib/tnl-neigh-cache.c
@@ -43,6 +43,12 @@
 #include "util.h"
 #include "openvswitch/vlog.h"
 
+#ifdef __sun
+#ifdef IFNAMSIZ
+#undef IFNAMSIZ
+#define IFNAMSIZ MAXLINKNAMESPECIFIER
+#endif
+#endif
 
 /* In seconds */
 #define NEIGH_ENTRY_DEFAULT_IDLE_TIME  (15 * 60)
diff --git a/ofproto/ofproto-dpif-xlate.c b/ofproto/ofproto-dpif-xlate.c
index a17047d2d..466fa12f6 100644
--- a/ofproto/ofproto-dpif-xlate.c
+++ b/ofproto/ofproto-dpif-xlate.c
@@ -67,6 +67,13 @@ COVERAGE_DEFINE(xlate_actions_too_many_output);
 
 VLOG_DEFINE_THIS_MODULE(ofproto_dpif_xlate);
 
+#ifdef __sun
+#ifdef IFNAMSIZ
+#undef IFNAMSIZ
+#define IFNAMSIZ MAXLINKNAMESPECIFIER
+#endif
+#endif
+
 /* Maximum depth of flow table recursion (due to resubmit actions) in a
  * flow translation.
  *
